src/main/resources/liquibase/crop_changelog/21_3_0.xml

```diff
@@ -0,0 +1,814 @@
+			delete from cvterm where cvterm_id = 8360;
+		</sql>
+	</changeSet>
+	<!-- End of SEED_SOURCE related changesets-->
+
+	<!-- ENTRY_TYPE related changesets-->
+	<changeSet author="lkovacic" id="v21.3.0-20">
+		<preConditions onFail="MARK_RAN">
+				<sqlCheck expectedResult="1">
+					SELECT count(1) FROM cvtermprop WHERE cvterm_id = 8255 AND type_id = 1800 AND value = 'Germplasm Descriptor';
+				</sqlCheck>
+		</preConditions>
+		<comment>Change 'ENTRY_TYPE' variable type to entry detail</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE cvtermprop SET value = 'Entry Detail' WHERE cvterm_id = 8255 AND type_id = 1800 AND value = 'Germplasm Descriptor';
+
+			INSERT INTO entry_detail_migration_output (cvterm_id) VALUES (8255);
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-21">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT count(1) > 0 FROM projectprop WHERE variable_id = 8255 AND type_id = 1804;
+			</sqlCheck>
+		</preConditions>
+		<comment>Change 'ENTRY_TYPE' variable type to entry detail in 'projectprop' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE projectprop SET type_id = 1815 WHERE variable_id = 8255 AND type_id = 1804;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-22">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<columnExists tableName="stockprop" columnName="cvalue_id"/>
+			</not>
+		</preConditions>
+		<comment>Add column 'cvalue_id' to 'stockprop' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			ALTER TABLE stockprop ADD cvalue_id INT NULL;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-23">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<columnExists tableName="stockprop" columnName="cvalue_id"/>
+				<sqlCheck expectedResult="0">
+					SELECT count(1) FROM stockprop WHERE cvalue_id IS NOT NULL;
+				</sqlCheck>
+				<sqlCheck expectedResult="1">
+					SELECT count(*) > 0
+					FROM (SELECT DISTINCT( sp.type_id ) AS typeId
+						  FROM stockprop sp
+								   INNER JOIN cvterm_relationship cr
+						  ON sp.type_id = cr.subject_id
+							  AND cr.type_id = 1220
+								   INNER JOIN cvterm_relationship cr2
+						  ON cr.object_id = cr2.subject_id
+							  AND cr2.type_id = 1190) AS inn;
+				</sqlCheck>
+			</and>
+		</preConditions>
+		<comment>Migrate categorical variable values to 'cvalue_id' column</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE stockprop sp
+			SET cvalue_id = value,
+				value     = (SELECT name
+							 FROM cvterm
+							 WHERE cvterm_id = cvalue_id)
+			WHERE value IN (SELECT cvterm_id
+							FROM cvterm_relationship cr
+									 INNER JOIN cvterm_relationship cr2 ON cr2.subject_id = cr.object_id AND cr2.type_id = 1190
+									 INNER JOIN cvterm c ON c.cvterm_id = cr2.object_id
+							WHERE cr.subject_id = sp.type_id AND cr.type_id = 1220) AND
+					type_id IN (SELECT inn.typeid
```

Comment:
@lik-83 same here, you can remove one select and subquery is not needed (same in the precondition) 
``
type_id IN (select DISTINCT(sp.type_id) AS typeId
									  FROM stockprop sp
											   INNER JOIN cvterm_relationship cr ON sp.type_id = cr.subject_id AND cr.type_id = 1220
											   INNER JOIN cvterm_relationship cr2 ON cr.object_id = cr2.subject_id AND cr2.type_id = 1190);
``

Reply:
In this case it is necessary because a derived table is needed to prevent a mysql error when updating the same table that it's being queried.